detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/Generator.kt

```diff
@@ -39,4 +39,35 @@ class Generator(
 
         outPrinter.println("\nGenerated all detekt documentation in $time ms.")
     }
+
+    fun executeCustomRuleConfig() {
+        val parser = KtCompiler()
+        val time = measureTimeMillis {
+            val ktFiles = arguments.inputPath
+                .flatMap { parseAll(parser, it) }
+
+            val folderMap = mutableMapOf<KtFile, String>()
+            ktFiles.forEach { ktFile ->
+                println(ktFile.name)
+                val pathParts = ktFile.name.split("/src/main/kotlin")
+                if (pathParts.count() > 1) {
+                    folderMap[ktFile] = "${pathParts.first()}/src/main/resources/config/"
+                }
+            }
+
+            folderMap
+                .toList()
+                .groupBy { (_, folder) -> folder }
+                .toList()
+                .forEach { (folder, list) ->
+                    val collector = DetektCollector()
+                    list.forEach { (file, _) ->
+                        collector.visit(file)
+                    }
+                    printer.printCustomRuleConfig(collector.items, folder)
+                }
```

Comment:
I think that the problem was that `ktFile.name.split("/src/main/kotlin")` on Windows should be `\src\main\kotlin`. But I think that it should not be necessary to use that. Instead of `flatMap` all the `KtFile`s we can keep them in different `List`s. I think that this could work:

```suggestion
            arguments.inputPath
                .map { parseAll(parser, it.resolve("src/main/kotlin/")) to it }
                .forEach { (list: Collection<KtFile>, folder: Path) ->
                    val collector = DetektCollector()
                    list.forEach { file ->
                        collector.visit(file)
                    }
                    printer.printCustomRuleConfig(
                        collector.items,
                        folder.resolve("src/main/resources/config/").toString()
                    )
                }
```